hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/sps/StoragePolicySatisfyManager.java

```diff
@@ -292,4 +292,13 @@ public boolean isEnabled() {
   public StoragePolicySatisfierMode getMode() {
     return mode;
   }
+
+  /**
+   * @return the number of paths to be processed by storage policy satisfier.
+   */
+  public int getPendingSPSPaths() {
+    synchronized (pathsToBeTraversed) {
+      return pathsToBeTraversed.size();
+    }
```

Comment:
This is synchronised and will be called a bunch of times. What impact is it going to have on normal SPS performance